### ISSUE TITLE
tooltip formatter when using axis trigger

### ISSUE THREAD

**Issue by ramarro123** [apache/incubator-echarts#9188]
hello,

if i use a 
 chart.tooltip = {
          trigger: 'axis'
        };

and

   chart.tooltip.formatter = function (params) {}

how can i detect what's the current item on params array?
params array have all the data from chart, so if i have 100 data, i have 100 items.

i am using a dataset+line with custom renderer here, and i need to have custom formatter as well.

let me know if i can detect the current index position (or the current X val from cursor, whatever better suit)

**Comment by Ovilia** [comment]
If you have 3 series of 100 data each, `params` will be an array of length 3, rather than 100. 
And since `trigger` is axis, we can't know which one of the three is the active one. In fact, we should say, all three of them is active. So it's expected behavior.

**Comment by ramarro123** [comment]
umm, i have created a demo here
http://jsfiddle.net/Lb4nkfcj/

i am using 4 series and a dataset with some "strange" mapping that will (eventually) allow me to create a gantt with a custom item renderer.

Having said so, now the chart doesn't display anything :) that's expected...

but, if i go around axe with cursor, i get "10" that's the size of dataset, not the size of series (that's 4)

on params, for tooltip, i get the full dataset, so it's impossible to know at which point of axes i am.
if i know the X position, then i can iterate the dataset to get the current position (or the nearest position, or whatever i want to do) but i haven't find a way to get the current X position.

even if trigger is axis (expecially with trigger axis i will say) having the position/value of axis it's necessary, it allow to understand the current position of cursor.

**Comment by ramarro123** [comment]
any update on that?

**Comment by 100pah** [comment]
(1) The 4 series is `type: 'custom'`, but I dont find a `renderItem` method, which is necessary in 'custom series'.
(2) When using `tooltip.trigger: 'axis'`, if we are hovering on a value like 'Pump' on y axis, all of the data item having 'Pump' on the dimension corresponding to y will be included in the `formatter` parameters.

**Comment by ramarro123** [comment]
1) even without renderItem method, grid should work (if you go mouseover on chart it should log the data on console i think)

2) if tooltip trigger = axis, and i go hovering, the formatter log ALL the data (10 entry in this case)
so how should i detect where am i?

**Comment by ramarro123** [comment]
ok, let me try again
http://jsfiddle.net/ap6ydveq/

this one have a renderitem, so you are happy, but still, the tooltip work in the same way :)

if you go on top of the item, X value it's the whole dataset source, making it impossible to detect where i am at the current position

**Comment by 100pah** [comment]
If the tooltip is `trigger: axis` and the axis is Y, only current y value can be obtained in the parameter of the tooltip.formatter.
If the x value is required, maybe change it to this one <http://jsfiddle.net/kucpfb8x/1/>.
